### ISSUE TITLE
RDS Create DBInstance From Snapshot

### ISSUE THREAD

**Issue by urton** [aws-controllers-k8s/community#1262]
Is there already a plan to allow creating a DBInstance resource from a snapshot similar to the restore-db-instance-from-db-snapshot via cli?

**Comment by jaypipes** [comment]
Hi @urton! No plan yet other than some randomish thoughts floating around in my head :) I will speak with the RDS team about this feature request and see how easily we can add it to the controller!

**Comment by urton** [comment]
Thanks for the quick response @jaypipes! Hopefully it is easy :) Same vein, what about managing pre-existing RDS instances via a DBInstance CR? Essentially, we want to start managing our RDS instances in K8s, but we want to avoid standing up instances/clusters from scratch and having to use DMS to migrate data.

**Comment by jaypipes** [comment]
@urton Well, for that second feature request, you're in luck, because we already support that :)

We have something called [`AdoptedResource`](https://aws-controllers-k8s.github.io/community/reference/common/v1alpha1/adoptedresource/). You can "ask" the ACK controller to take over management of an existing resource by calling `kubectl create` and passing in an ACK AdoptedResource CR, like so:

```bash
cat <<EOF >/my_adopted_db.yaml
apiVersion: services.k8s.aws/v1alpha1
kind: AdoptedResource
metadata:
  name: my-adopted-db
spec:
  aws:
    arn: your-rds-db-instance-arn
  kubernetes:
    group: rds.services.k8s.aws
    kind: DBInstance
EOF
kubectl create -f my_adopted_db.yaml
```

@RedbackThomson can tell you more about the AdoptedResource functionality.

**Comment by urton** [comment]
@jaypipes that is exactly what I was looking for. THANK YOU. I did have to add the spec.aws.nameOrID, but that was easy engouh to find and not I've adopted an instance or two.. This will help our immediate transition, but we do create instances from snapshots frequently, so that option would be pretty sweet to have as well sometime in the near future.

Separately, I think it would be really helpful to have a status column in the get dbinstance response without having to do a custom-columns output with 'STATUS:{.status.dbInstanceStatus}'. I can create a new enhancement request for that if you'd like.

Thanks again for the prompt replies and getting me squared away!
